### ISSUE TITLE
Nimbus VC to Lodestar v1.8.0 BN attestation errors

### ISSUE THREAD

**Issue by philknows** [ChainSafe/lodestar#5496]
**Describe the bug**

As mentioned by RockLogic in Discord: https://discord.com/channels/593655374469660673/1077978309775863900/1107299213634711552

connecting a Nimbus VC to a Lodestar BN creates attestation issues.

```ts
#150:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:23)
#149:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
#148:     at Object.getAggregatedAttestation (file:///usr/app/packages/beacon-node/src/api/impl/validator/index.ts:509:37)
#147:     at AttestationPool.getAggregate (file:///usr/app/packages/beacon-node/src/chain/opPools/attestationPool.ts:136:13)
#146: Error: No attestation for slot=6437249 dataRoot=0x02449b5da7b01ab911dfcfb8deffe08348be29f12f170763897de5c9d23b760c
#145: May-14 13:30:27.504[rest]            [31merror[39m: Req req-2nvf getAggregatedAttestation error  No attestation for slot=6437249 dataRoot=0x02449b5da7b01ab911dfcfb8deffe08348be29f12f170763897de5c9d23b760c
#144:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:23)
#143:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
#142:     at Object.getAggregatedAttestation (file:///usr/app/packages/beacon-node/src/api/impl/validator/index.ts:509:37)
#141:     at AttestationPool.getAggregate (file:///usr/app/packages/beacon-node/src/chain/opPools/attestationPool.ts:136:13)
#140: Error: No attestation for slot=6437249 dataRoot=0x795ed69cdc12896e4c20047445d7a76239fe8612b14f5a7179702edd619c23bc
#139: May-14 13:30:27.503[rest]            [31merror[39m: Req req-2nve getAggregatedAttestation error  No attestation for slot=6437249 dataRoot=0x795ed69cdc12896e4c20047445d7a76239fe8612b14f5a7179702edd619c23bc
#138:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#137:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#136:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#135:     at Array.map (<anonymous>)
#134:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#133:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#132:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#131:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#130:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#129:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#128: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#127: May-14 13:30:26.359[rest]            [31merror[39m: Req req-2nva submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xb6cfa4b0130a3c99be1129938a250e9ed050f78b09c901ac67bb9bd61f3a4fb6
#126:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#125:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#124:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#123:     at Array.map (<anonymous>)
#122:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#121:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#120:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#119:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#118:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#117:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#116: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#115: May-14 13:30:26.359[rest]            [31merror[39m: Req req-2nv9 submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xb6cfa4b0130a3c99be1129938a250e9ed050f78b09c901ac67bb9bd61f3a4fb6
#114:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#113:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#112:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#111:     at Array.map (<anonymous>)
#110:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#109:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#108:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#107:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#106:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#105:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#104: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#103: May-14 13:30:26.358[api]             [31merror[39m: Error on submitPoolAttestations [0] slot=6437249, index=41 code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xb6cfa4b0130a3c99be1129938a250e9ed050f78b09c901ac67bb9bd61f3a4fb6
#102:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#101:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#100:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#99:     at Array.map (<anonymous>)
#98:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#97:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#96:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#95:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#94:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#93:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#92: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#91: May-14 13:30:26.357[api]             [31merror[39m: Error on submitPoolAttestations [0] slot=6437249, index=41 code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xb6cfa4b0130a3c99be1129938a250e9ed050f78b09c901ac67bb9bd61f3a4fb6
#90: May-14 13:30:18.816[]                 [32minfo[39m: Synced - slot: 6437249 - head: (slot -19) 0xf27e…2c4a - exec-block: valid(17258382 0x5cd8…) - finalized: 0xe6e2…7140:201161 - peers: 41
#89:     at processTicksAndRejections (node:internal/process/task_queues:82:21)
#88:     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
#87:     at emitErrorNT (node:internal/streams/destroy:151:8)
#86:     at Socket.emit (node:events:513:28)
#85:     at Socket.socketErrorListener (node:_http_client:502:9)
#84:     at ClientRequest.emit (node:events:513:28)
#83:     at ClientRequest.<anonymous> (/usr/app/node_modules/node-fetch/lib/index.js:1461:11)
#82: FetchError: request to http://stereum-42014ab4-219d-1630-a12a-cd0c4276b443:18550/eth/v1/builder/validators failed, reason: write EPIPE
#81: May-14 13:30:18.418[rest]            [31merror[39m: Req req-2nte registerValidator error  request to http://stereum-42014ab4-219d-1630-a12a-cd0c4276b443:18550/eth/v1/builder/validators failed, reason: write EPIPE
#80:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:23)
#79:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
#78:     at Object.getAggregatedAttestation (file:///usr/app/packages/beacon-node/src/api/impl/validator/index.ts:509:37)
#77:     at AttestationPool.getAggregate (file:///usr/app/packages/beacon-node/src/chain/opPools/attestationPool.ts:136:13)
#76: Error: No attestation for slot=6437248 dataRoot=0x5270d03d0d5d95f021821893fc997a6bce9d2d98ae605ea6d348f1ec100766a3
#75: May-14 13:30:13.706[rest]            [31merror[39m: Req req-2nui getAggregatedAttestation error  No attestation for slot=6437248 dataRoot=0x5270d03d0d5d95f021821893fc997a6bce9d2d98ae605ea6d348f1ec100766a3
#74:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:23)
#73:     at processTicksAndRejections (node:internal/process/task_queues:95:5)
#72:     at Object.getAggregatedAttestation (file:///usr/app/packages/beacon-node/src/api/impl/validator/index.ts:509:37)
#71:     at AttestationPool.getAggregate (file:///usr/app/packages/beacon-node/src/chain/opPools/attestationPool.ts:136:13)
#70: Error: No attestation for slot=6437248 dataRoot=0xf38866ffae4d65dc801cb97de9910055639bd059d29e80a48736e8e74d9d8e5e
#69: May-14 13:30:13.706[rest]            [31merror[39m: Req req-2nuh getAggregatedAttestation error  No attestation for slot=6437248 dataRoot=0xf38866ffae4d65dc801cb97de9910055639bd059d29e80a48736e8e74d9d8e5e
#68:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#67:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#66:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#65:     at Array.map (<anonymous>)
#64:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#63:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#62:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#61:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#60:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#59:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#58: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#57: May-14 13:30:12.955[rest]            [31merror[39m: Req req-2nuc submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xa5a40ac90a9ca56e1b49523437c3c00ddb0863d6020efb3567f6ccc682851332
#56:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#55:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#54:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#53:     at Array.map (<anonymous>)
#52:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#51:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#50:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#49:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#48:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#47:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#46: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#45: May-14 13:30:12.955[rest]            [31merror[39m: Req req-2nub submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xa5a40ac90a9ca56e1b49523437c3c00ddb0863d6020efb3567f6ccc682851332
#44:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#43:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#42:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#41:     at Array.map (<anonymous>)
#40:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#39:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#38:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#37:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#36:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#35:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#34: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#33: May-14 13:30:12.955[rest]            [31merror[39m: Req req-2nua submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xa5a40ac90a9ca56e1b49523437c3c00ddb0863d6020efb3567f6ccc682851332
#32:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#31:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#30:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#29:     at Array.map (<anonymous>)
#28:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#27:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#26:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#25:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#24:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#23:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#22: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#21: May-14 13:30:12.955[rest]            [31merror[39m: Req req-2nu9 submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xa5a40ac90a9ca56e1b49523437c3c00ddb0863d6020efb3567f6ccc682851332
#20:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#19:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#18:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#17:     at Array.map (<anonymous>)
#16:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#15:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#14:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#13:     at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
#12:     at verifyHeadBlockAndTargetRoot (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:343:21)
#11:     at verifyHeadBlockIsKnown (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:376:11)
#10: Error: ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT
#9: May-14 13:30:12.955[rest]            [31merror[39m: Req req-2nu8 submitPoolAttestations error  code=ATTESTATION_ERROR_UNKNOWN_OR_PREFINALIZED_BEACON_BLOCK_ROOT, root=0xa5a40ac90a9ca56e1b49523437c3c00ddb0863d6020efb3567f6ccc682851332
#8:     at preHandlerCallback (/usr/app/node_modules/fastify/lib/handleRequest.js:128:37)
#7:     at Object.handler (file:///usr/app/packages/api/src/utils/server/genericJsonServer.ts:45:41)
#6:     at Object.submitPoolAttestations (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:53:22)
#5:     at Array.map (<anonymous>)
#4:     at file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:61:72
#3:     at validateGossipFnRetryUnknownRoot (file:///usr/app/packages/beacon-node/src/network/processor/gossipHandlers.ts:403:20)
#2:     at validateFn (file:///usr/app/packages/beacon-node/src/api/impl/beacon/pool/index.ts:56:38)
#1: at validateGossipAttestation (file:///usr/app/packages/beacon-node/src/chain/validation/attestation.ts:143:26)
```

**Expected behavior**

Should be able to properly interop with the Nimbus VC.

**Comment by twoeths** [comment]
Lodestar v1.8 already handle unknown root attestation by waiting for the block to come. However it only waits for max 1 slot, if block comes after that it still throw the error. Two things could fix this issue:
- [x] network thread which help block to be processed faster consistently, to be released in v1.9
- [x] trigger block search when we receive unknown root attestation, PR in progress: https://github.com/ChainSafe/lodestar/pull/5485

**Comment by twoeths** [comment]
this should be fixed with v1.9.0

**Comment by nflaig** [comment]
Can confirm that this is fixed with v1.9.0, been running Nimbus VC for two days now, performance is the same to Lodestar VC and there are no errors on the beacon node.
